vuepress/docs/docs/middleware.md

```diff
@@ -154,4 +156,37 @@ a federated graph:
       def apply[R1](original: GraphQL[R1]): GraphQL[R1] =
         federate(original)
     }
-```
\ No newline at end of file
+```
+
+## Cost Estimation
+
+The `queryCost` and `maxCost` wrappers as well as their variants can be used to estimate the cost of a query, however they require a bit 
+more set up to work properly.
+
+These wrappers are in the `CostEstimation` object which also comes with a special directive that can be used out of the box to instrument your
+schema for cost analysis.
+
+Given a schema in which different fields have different costs to execute, either because they require additional network or computing resources, or database access, or they 
+have some other dependency that makes them expensive to compute. You can add the `CostDirective` to your resolver like so:
+
+```scala
+case class SpokenLineArgs(offset: Int, limit: Int)
+
+@GQLDirective(CostDirective(2))
```

Comment:
Is it possible to specify a directive in `CostEstimation`, like `GQLCost` that desugars to `@GQLDirective(CostDirective(2))`? I'm thinking that if you use this in an API this will be used very frequently in your schema so saving some typing might be worth it.

Reply:
Hmm interesting idea, I'll look into if this is possible.

Reply:
Also use the new apply in the docs